l5kit/l5kit/rasterization/semantic_rasterizer.py

```diff
@@ -1,73 +1,53 @@
-        raster_size (Tuple[int, int]): Size of the image to render.
-        pixel_size (np.ndarray): The scale of one pixel in the real world (in meters).
+        center (float): XY of the center
+        bounds (np.ndarray): array of shape Nx2x2 [[x_min,y_min],[x_max, y_max]]
+        half_extent (float): half the side of the bounding box centered around center
 
     Returns:
-        np.ndarray: RGB raster.
+        np.ndarray: indices of elements inside radius from center
     """
+    x_center, y_center = center
 
-    img = 255 * np.ones(shape=(raster_size[1], raster_size[0], 3), dtype=np.uint8)
-
-    def map_to_image(px: np.ndarray, py: np.ndarray) -> Tuple[np.ndarray, np.ndarray]:
-        rx = (np.cos(yaw) * (px - x) - np.sin(yaw) * (py - y)) / pixel_size[0] + 0.5 * raster_size[0]
-        ry = (np.sin(yaw) * (px - x) + np.cos(yaw) * (py - y)) / pixel_size[1] + 0.5 * raster_size[1]
-        return rx.astype(int) * 256, ry.astype(int) * 256
-
-    def elements_within_radius(bounding_box: dict, r: np.ndarray) -> np.ndarray:
-        return np.nonzero(
-            (x > bounding_box["min_x"] - r)
-            & (x < bounding_box["max_x"] + r)
-            & (y > bounding_box["min_y"] - r)
-            & (y < bounding_box["max_y"] + r)
-        )[0]
-
-    r = np.linalg.norm(raster_size * pixel_size)
-
-    # plot lanes
-    lines = []
-    for i in elements_within_radius(semantic_map["lanes_bounds"], r):
-        lane = semantic_map["lanes"][i]
-        x1, y1 = map_to_image(lane[0][0], lane[0][1])
-        x2, y2 = map_to_image(lane[1][0], lane[1][1])
-        pts1 = np.vstack((x1, y1)).T
-        pts2 = np.vstack((x2, y2)).T
-        lines.append(pts1)
-        lines.append(pts2)
-        poly = np.vstack((pts1, np.flip(pts2, 0)))
-        cv2.fillPoly(img, [poly], (17, 17, 31), lineType=cv2.LINE_AA, shift=8)
+    x_min_in = x_center > bounds[:, 0, 0] - half_extent
+    y_min_in = y_center > bounds[:, 0, 1] - half_extent
+    x_max_in = x_center < bounds[:, 1, 0] + half_extent
+    y_max_in = y_center < bounds[:, 1, 1] + half_extent
+    return np.nonzero(x_min_in & y_min_in & x_max_in & y_max_in)[0]
 
-    cv2.polylines(img, lines, False, (255, 217, 82), lineType=cv2.LINE_AA, shift=8)
 
-    # plot crosswalks
-    crosswalks = []
-    for i in elements_within_radius(semantic_map["crosswalks_bounds"], r):
-        crosswalk = semantic_map["crosswalks"][i]
-        x1, y1 = map_to_image(crosswalk[0], crosswalk[1])
-        pts = np.vstack((x1, y1)).T
-        crosswalks.append(pts)
+def cv2_subpixel(coords: np.ndarray) -> np.ndarray:
+    """
+    Cast coordinates to numpy.int but keep fractional part by previously multiplying by 2**CV2_SHIFT
 
-    cv2.polylines(img, crosswalks, True, (255, 117, 69), lineType=cv2.LINE_AA, shift=8)
+    Args:
+        coords (np.ndarray): XY coords as float
 
-    return img
+    Returns:
+        np.ndarray: XY coords as int for cv2 shift draw
+    """
+    coords = coords * 2 ** CV2_SHIFT
```

Comment:
Minor nit: Can we save the `2 ** CV2_SHIFT` as a constant so we don't have to do it every time the function is called?